Source/WebCore/html/HTMLSlotElement.cpp

```diff
@@ -165,18 +166,26 @@ Vector<Ref<Element>> HTMLSlotElement::assignedElements(const AssignedNodesOption
     });
 }
 
-void HTMLSlotElement::assign(FixedVector<std::reference_wrapper<Node>>&& nodes)
+void HTMLSlotElement::assign(FixedVector<ElementOrText>&& nodes)
 {
     for (auto& node : m_manuallyAssignedNodes)
         node->setManuallyAssignedSlot(nullptr);
 
     auto previous = std::exchange(m_manuallyAssignedNodes, { });
-    HashSet<Ref<Node>> seenNodes;
-    m_manuallyAssignedNodes = WTF::compactMap(nodes, [&](Node& node) -> std::optional<WeakPtr<Node>> {
-        if (seenNodes.contains(node))
-            return std::nullopt;
-        seenNodes.add(node);
-        return WeakPtr { node };
+    HashSet<RefPtr<Node>> seenNodes;
+    m_manuallyAssignedNodes = WTF::compactMap(nodes, [&seenNodes](ElementOrText& node) -> std::optional<WeakPtr<Node>> {
+        auto mapper = [&seenNodes]<typename T>(RefPtr<T>& node) -> std::optional<WeakPtr<Node>> {
+            // RefPtr<Node>& node = derived;
```

Comment:
What is this comment about? Remove?

Reply:
I removed it in https://github.com/WebKit/WebKit/pull/3215/commits/ef32b1f3be32c721015225b5c02d3bfba44dde08